src/python/test/test_dxpy.py

```diff
@@ -1877,6 +1877,23 @@ def test_retry_after_without_header_set(self):
         self.assertTrue(50000 <= time_elapsed)
         self.assertTrue(time_elapsed <= 70000)
 
+    @unittest.skip("Example of a request that should be retried")
+    def test_bad_host(self):
+        # This call shouldn't return immediately, rather it should be
+        # retried with the entire retry loop taking approximately 1
+        # minute.
+        dxpy.DXHTTPRequest('http://doesnotresolve.dnanexus.com/', {}, prepend_srv=False, always_retry=False)
+
+    @unittest.skip("Example of a request that should be retried")
+    def test_connection_refused(self):
+        # This call shouldn't return immediately, rather it should be
+        # retried with the entire retry loop taking approximately 1
+        # minute.
+        #
+        # Connecting to a port on which there is no server running
+        dxpy.DXHTTPRequest('http://localhost:20406', {}, prepend_srv=False, always_retry=False)
```

Comment:
Indeed, one way out may also be to keep the retry behavior the same but to print name resolution failures to stderr.